src/components/accordion/accordion.tsx

```diff
@@ -150,42 +149,36 @@ export class EuiAccordion extends Component<
     );
 
     const icon = (
-      <EuiIcon type={this.state.isOpen ? 'arrowDown' : 'arrowRight'} size="m" />
+      <EuiIcon
+        className="euiAccordion__icon"
+        type={this.state.isOpen ? 'arrowDown' : 'arrowRight'}
+        size="m"
+      />
     );
 
     let optionalAction = null;
 
     if (extraAction) {
-      optionalAction = <EuiFlexItem grow={false}>{extraAction}</EuiFlexItem>;
+      optionalAction = (
+        <div className="euiAccordion__optionalAction">{extraAction}</div>
+      );
     }
 
     return (
       <div className={classes} {...rest}>
-        <EuiFlexGroup gutterSize="none" alignItems="center">
-          <EuiFlexItem>
-            <button
-              aria-controls={id}
-              aria-expanded={!!this.state.isOpen}
-              onClick={this.onToggle}
-              className={buttonClasses}
-              type="button">
-              <EuiFlexGroup
-                gutterSize="s"
-                alignItems="center"
-                responsive={false}>
-                <EuiFlexItem grow={false} className="euiAccordion__iconWrapper">
-                  {icon}
-                </EuiFlexItem>
-
-                <EuiFlexItem className={buttonContentClasses}>
-                  {buttonContent}
-                </EuiFlexItem>
-              </EuiFlexGroup>
-            </button>
-          </EuiFlexItem>
+        <div className="euiAccordion__triggerWrapper">
+          <button
+            aria-controls={id}
+            aria-expanded={!!this.state.isOpen}
+            onClick={this.onToggle}
+            className={buttonClasses}
+            type="button">
+            <span className="euiAccordion__iconWrapper">{icon}</span>
+            <span className={buttonContentClasses}>{buttonContent}</span>
```

Comment:
The `buttonContentClasses` contains the `buttonContentClassName` prop and `euiAccordion__buttonContent` class, which I can't find in the existing styles. Could we get rid of `buttonContentClassName` at all since we can pass  a `buttonContent` node with already applied class ? And also get rid of `buttonContentClasses` ..

```suggestion
            {buttonContent}
```

Reply:
I see your point about the class `.euiAccordion__buttonContent` not being used in any SASS, I can remove that class. However, we need to keep `buttonContentClassName` because as a consumer, you can't target that inner span wihout overriding a `.eui` class. 

So we're providing a way to target that exact span, which needs to happen if you need to truncate the text, as is shown in this example:  https://github.com/elastic/eui/blob/92282d59314ab604dcb225f454b52d79c2879f77/src-docs/src/views/accordion/accordion_multiple.js#L23

<img width="375" alt="Screen Shot 2019-07-22 at 09 38 56 AM" src="https://user-images.githubusercontent.com/549577/61636884-8c2d5900-ac64-11e9-8803-cd8f418396b9.png">

Reply:
What is a benefit of providing a className for a node which I also provide manually? That means I could pass the `buttonContent` prop as node : `<span className="eui-textTruncate">My very very very long text is here</span>`

Reply:
Because not all consumers know to do it that way, haha